docs/VERSIONING.rst

```diff
@@ -0,0 +1,694 @@
+including null versions and delete markers, described in the above
+links.
+
+Implementation of Bucket Versioning in S3
+-----------------------------------------
+
+Overview of Metadata and API Component Roles
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each version of an object is stored as a separate key in metadata. The
+S3 API interacts with the metadata backend to store, retrieve, and
+delete version metadata.
+
+The implementation of versioning within the metadata backend is naive.
+The metadata backend does not evaluate any information about bucket or
+version state (whether versioning is enabled or suspended, and whether a
+version is a null version or delete marker). The S3 front-end API
+manages the logic regarding versioning information, and sends
+instructions to metadata to handle the basic CRUD operations for version
+metadata.
+
+The role of the S3 API can be broken down into the following:
+
+-  put and delete version data
+-  store extra information about a version, such as whether it is a
+   delete marker or null version, in the object's metadata
+-  send instructions to metadata backend to store, retrieve, update and
+   delete version metadata based on bucket versioning state and version
+   metadata
+-  encode version ID information to return in responses to requests, and
+   decode version IDs sent in requests
+
+The implementation of Bucket Versioning in S3 is described in this
+document in two main parts. The first section, `"Implementation of
+Bucket Versioning in
+Metadata" <#implementation-of-bucket-versioning-in-metadata>`__,
+describes the way versions are stored in metadata, and the metadata
+options for manipulating version metadata.
+
+The second section, `"Implementation of Bucket Versioning in
+API" <#implementation-of-bucket-versioning-in-api>`__, describes the way
+the metadata options are used in the API within S3 actions to create new
+versions, update their metadata, and delete them. The management of null
+versions and creation of delete markers are also described in this
+section.
+
+Implementation of Bucket Versioning in Metadata
+-----------------------------------------------
+
+As mentioned above, each version of an object is stored as a separate
+key in metadata. We use version identifiers as the suffix for the keys
+of the object versions, and a special version (the `"Master
+Version" <#master-version>`__) to represent the latest version.
+
+An example of what the metadata keys might look like for an object
+``foo/bar`` with three versions:
+
++-------------------------------------------------------------------+
+| key                                                               |
++===================================================================+
+| foo/bar                                                           |
++-------------------------------------------------------------------+
+| foo/bar:raw-latex:`\0`98506163554375999999PARIS 0.a430a1f85c6ec   |
```

Comment:
I'm not sure if the raw-latex is intentional formatting but it doesn't seem to be working, I see it in the rendered output (http://s3-server.readthedocs.io/en/doc-adddoc/VERSIONING/)
And the `\` still seems interpreted, maybe you can put two to escape it? `\\`